tensorboard/plugins/graph/tf_graph_loader/tf-graph-loader.html

```diff
@@ -81,18 +83,33 @@
     },
   },
   observers: [
-    '_selectedDatasetChanged(selectedDataset, datasets, overridingHierarchyParams, compatibilityProvider)',
+    '_selectionChanged(selection, overridingHierarchyParams, compatibilityProvider)',
     '_selectedFileChanged(selectedFile, overridingHierarchyParams, compatibilityProvider)',
-    '_readAndParseMetadata(selectedMetadataTag, overridingHierarchyParams)',
   ],
-  _readAndParseMetadata: function(metadataIndex) {
-    if (metadataIndex == -1 || this.datasets[this.selectedDataset] == null ||
-        this.datasets[this.selectedDataset].runMetadata == null ||
-        this.datasets[this.selectedDataset].runMetadata[metadataIndex] == null) {
+  _selectionChanged() {
+    // selection can change a lot within a microtask.
+    // Don't fetch too much too fast and introduce race condition.
+    this.debounce('selectionchange', () => this._load());
+  },
+  _load: function() {
+    const {selection, overridingHierarchyParams, compatibilityProvider} = this;
```

Comment:
`compatibilityProvider` unused? (Also, I don’t know what it is, but I’ll
take your word for it if you say that it’s not too important right now.)

Reply:
You are right. `compatibilityProvider` is used in one of the method we call here.